game_patch/input/key.cpp

```diff
@@ -461,21 +461,35 @@ CodeInjection item_touch_weapon_autoswitch_patch{
     }
 };
 
-CodeInjection key_down_handler_injection{
-    0x0051E9AC,
-    [] (auto& regs) {
-        const int virtual_key = addr_as_ref<int>(regs.esp + 4);
-        // For numeric keypads, we need to fix these keys' scan codes.
-        auto& scan_code = regs.eax;
-        if (virtual_key == VK_PRIOR) {
-            scan_code = rf::KEY_PAGEUP;
-        } else if (virtual_key == VK_NEXT) {
-            scan_code = rf::KEY_PAGEDOWN;
-        } else if (virtual_key == VK_END) {
-            scan_code = rf::KEY_END;
-        } else if (virtual_key == VK_HOME) {
-            scan_code = rf::KEY_HOME;
+FunHook<void(int, int, int)> key_msg_handler_hook{
+    0x0051EBA0,
+    [] (const int msg, const int w_param, int l_param) {
+        // For num pads, we need to fix these scan codes.
+        switch (msg) {
+            case WM_KEYDOWN:
+            case WM_SYSKEYDOWN:
+            case WM_KEYUP:
+            case WM_SYSKEYUP: {
+                uint32_t scan_code = (l_param >> 16) & 0xFF;
+                constexpr uint32_t EXTENDED_KEY_FLAG = 0x1000000;
```

Comment:
Does the extended key flag need to be cleared? We're already rewriting the scancodes, is there a need to clear the flag? It probably doesn't matter, but if it's not necessary it would be good to keep it, in case we need to differentiate between the two variants at any point down the road.

Additionally, couldn't we use the extended key flag in this block to only rewrite the num pad versions? I'm not sure off-hand if we'd expect the extended flag to be set or not for num pad vs. nav block though.